### PR TITLE
Onboarding Simplified version without maps location

### PR DESCRIPTION
**Pull request by jlsilva1994**
Changes/features included in this commit:
 - Added dependencies in gradle for Google maps API
 - Added permissions to use internet, gps and storage ; and registry new activities in the android Manifest
 - Changed the layout and code of HomeActivity to support a button to open Onboarding activity
 - Added the layout files and Activity files for OnboardingActivity , LeisureAppsActivity and WorkRelatedAppsActivity

Onboarding Activity contains methods to define and store in the sharedPreferences the Work Shesdule, choose the work related and the leisure applications from a list of all Apps intalled.
The option to choose your work location and home location is NOT fully implemented: the API is working, the maps are created but it is not implemented a way to manipulate them yet!

ContextRelatedAppsActivity implements the list where to choose and store the Context related Applications
 AppCheckAdapter is an auxiliary class to implement the listView adapter for choose the applications

UPDATE:
- added method to support maps manipulation
- changed onboarding layout, now using tab views to better organization and to fix an issue with maps and scrolling OnboardingActivity

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,3 +54,10 @@ google-services.json
 freeline.py
 freeline/
 freeline_project_description.json
+
+# MAC OS generated files
```
Comment (martinhoaragao) [inline .gitignore]: I'd avoid putting these into a gitignore, the mac user should have them in his own global file.

File: .idea/modules.xml [hunk 2]
```diff
@@ -2,10 +2,8 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/Adamastor.iml" filepath="$PROJECT_DIR$/Adamastor.iml" />
```
Comment (martinhoaragao) [inline .idea/modules.xml]: What does this change? Issues with case insensitive?

**Review by martinhoaragao — CHANGES_REQUESTED** [review]
This is quite a big chunk of changes.

Can you add a small sum up on the description? 16 files with 822 lines added. With a shallow review we will most likely miss things that you added with a purpose.

File: .gitignore (hunk 1 above)
Comment (jlsilva1994) [reply .gitignore]: ok, i will change that.

File: .idea/modules.xml (hunk 2 above)
Comment (jlsilva1994) [reply .idea/modules.xml]: yes

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -3,33 +3,69 @@
     package="corp.kairos.adamastor">
 
     <uses-permission android:name="android.permission.VIBRATE" />
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (hgg) [inline app/src/main/AndroidManifest.xml]: Why do we need external storage access?

Comment (jlsilva1994) [reply app/src/main/AndroidManifest.xml]: its for the google's API to work properly

**Comment by martinhoaragao** [comment]
Is this ready to be re-reviewed? Since you have added quite a few commits afterward.

**Comment by jlsilva1994** [comment]
yes Martinho is ready, i did a rebase from develop last changes and squash some commits from last update

**Comment by djcouto** [comment]
I tested and it's working fine, at this point i'm not worried about implementation alternatives, so for me is good to go :+1:

**Review by FranciscoRibeiro — CHANGES_REQUESTED** [review]
Setting the necessary information during the onboarding process seems to be working fine (time and apps).

However, I can't save the changes because the button on the bottom is not visible on my device's screen. 
I guess it has to do with its position not being set in a "relative" way? Is that the reason why it's having issues showing properly?

File: app/src/main/res/layout/onboarding.xml [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                android:id="@android:id/tabs"
+                android:layout_width="match_parent"
+                android:layout_height="45dp">
+
+            </TabWidget>
+
+            <FrameLayout
+                android:id="@android:id/tabcontent"
+                android:layout_width="match_parent"
+                android:layout_height="450dp">
```
Comment (FranciscoRibeiro) [inline app/src/main/res/layout/onboarding.xml]: I think you want to put `android:layout_height="wrap_content">` here. Does this fix the bottom button issue?

Comment (jlsilva1994) [reply app/src/main/res/layout/onboarding.xml]: yes, it helped. thanks for the heads up! i changed the layout and fixed the issue

File: app/src/main/java/corp/kairos/adamastor/Onboarding/OnboardingActivity.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.GregorianCalendar;
+
+import corp.kairos.adamastor.HomeActivity;
+import corp.kairos.adamastor.R;
+import corp.kairos.adamastor.Settings;
+
+public class OnboardingActivity extends AppCompatActivity {
```
Comment (djcouto) [inline app/src/main/java/corp/kairos/adamastor/Onboarding/OnboardingActivity.java]: Why `AppCompatActivity`?
https://stackoverflow.com/questions/31297246/activity-appcompatactivity-fragmentactivity-and-actionbaractivity-when-to-us

**Review by djcouto — COMMENTED** [review]
The result is fantastic, I particularly like the way of picking the time. 
For me its good enough to be approved. I have only one suggestion. 
In actual approach it's possible to choose a final time before the initial time, can you fix it? I assume it's not very difficult because the class `TimePickerDialog` has a `setMinTime()`  method, at least by the documentation. 
Either way, it would be nice if you fix this, with the interface, or with a verification when saving new changes.
Nevertheless, great job mate :+1:

**Comment by dfvt** [comment]
@djcouto It should be possible to set a final time before initial time because a person can work from 21h until 4h

**Comment by djcouto** [comment]
@dfvt wow shit your right, my bad. Approved than.

**Review by djcouto — APPROVED** [review]

**Pull request merged**